### DISCUSSION TITLE
Expo Support using expo-clipboard?

### DISCUSSION THREAD

**Discussion by crosstechnology** [tttstudios/react-native-otp-input#113] [Ideas]
Would it be possible for this module to support expo by way of allowing for implementation of https://docs.expo.io/versions/latest/sdk/clipboard/ as an alternative to rnc-clipboard?

Or is there a fundamental feature missing from the expo-clipboard that renders it useless for this purpose?

**Comment by RayBans2111** [comment]
Hey cross, I wanted to know if you found a workaround for the time being? Or did you use a different library to make an otp input?

**Reply by crosstechnology** [reply]
Hey @RayBans2111,

I didn't end up using this library. Instead I just used an **Input** object from **react-native-elements**, and what I found was that as long as:

1. My Input contained the key **textContentType="oneTimeCode"**
2. My OTP SMS used the phrase **verification code**

Then the Input element of type oneTimeCode would automatically support the quick-fill from the keyboard (iOS) and the copy option from the notification bubble (Android)

So a sample OTP SMS would be something like:

```
Hi RayBans2111,

Your Ray Bans app verification code is: 123456
```

And a sample Input element would be something like:
```
<Input
    textContentType="oneTimeCode"
    inputStyle={{
        color: "#000",
        textAlign: "center",
        fontSize: 30,
    }}
    placeholder="- - - - - -"
    placeholderTextColor="#999"
    maxLength={6}
    keyboardType="number-pad"
    onChangeText={ (otpVal) => { setOTP(otpVal);} }
    value={otp}
    onSubmitEditing={() => {
        if (otp.length === 6) {
            //doSomething
        }
    }}
/>
```

I hope this helps!

**Reply by RayBans2111** [reply]
Oh cool, thanks for the helpful tip. I really appreciate that!
